### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Juanse7793**
## Requirements

- Create a table named `vets` with the following columns: :+1: 
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
  - `age`: integer
  - `date_of_graduation`: date
- There is a **many-to-many** relationship between the tables `species` and `vets`: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called `specializations` to handle this relationship. :100: 

- There is a **many-to-many** relationship between the tables `animals` and `vets`: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called `visits` to handle this relationship, it should also keep track of the date of the visit. :100: 

- Insert the following data for vets: :heavy_check_mark: 
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- Insert the following data for specialties: :heavy_check_mark: 
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.
- Insert the following data for visits: :heavy_check_mark: 
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.
- Write queries to answer the following: :nerd_face: 
  - Who was the last animal seen by William Tatcher?
  - How many different animals did Stephanie Mendez see?
  - List all vets and their specialties, including vets with no specialties.
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  - What animal has the most visits to vets?
  - Who was Maisy Smith's first visit?
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many visits were with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -46,4 +46,65 @@ SET owners_id =
...
+  END;
+
+--   Insert the following data for vets:
+-- Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
+INSERT INTO vets VALUES (DEFAULT, 'William Tatcher', 45, '2000-04-23');
+-- Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
+INSERT INTO vets VALUES (DEFAULT, 'Maisy Smith', 26, '2019-01-17');
+-- Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
+INSERT INTO vets VALUES (DEFAULT, 'Stephanie Mendez', 64, '1981-05-04');
+-- Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
```
Comment (mmsesay) [inline data.sql:58]: - [x] I guess you forgot to insert the record for vet `Jack Harknes`

```sql
 INSERT INTO vets (name, age, date_of_graduation)
 VALUES ('Jack Harknes', 38, '2008-06-08');
```

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @Juanse7793, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (Juanse7793) [reply data.sql:58]: thank you for the feedback :handshake: I put him on my database but I forgot to write the line of code.

**Review by keneogu — APPROVED** [review]
> ## Hi @Juanse7793,
> Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
> 
> ## Congratulations!     
> Cheers and Happy coding!clapclapclap
> 
> image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966) image [ ](https://camo.githubusercontent.com/2366be437d35bce33dd123620530e9131a8ac147fe960e76a36614dc703e8e22/68747470733a2f2f692e70696e696d672e636f6d2f6f726967696e616c732f37372f30652f62302f37373065623035373434393538326334343637326232643264316363333962362e676966)
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**